### PR TITLE
add Twitch unit test - it closes (#121)

### PR DESCRIPTION
**Pull request by ekafyi**
PR for https://github.com/PaulieScanlon/mdx-embed/issues/121

For the IntersectionObserver, I used the code from https://github.com/PaulieScanlon/mdx-embed/blob/main/packages/mdx-embed/src/components/general-observer/general-observer.test.tsx

Let me know if there's anything to modify!

### CODE REVIEW

File: packages/mdx-embed/src/components/twitch/twitch.test.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { render, screen, act } from '@testing-library/react';
+
+import { Twitch } from './';
+
+const PARENT_PROP = 'http://localhost:6006/';
+
+describe('<Twitch />', () => {
+  const intersectionObserverMock = () => ({
```
Comment (PaulieScanlon) [inline packages/mdx-embed/src/components/twitch/twitch.test.tsx:9]: Hey @ekafyi  thanks for this. The `intersectionObserverMock` used in General Observer is a special case. There's a more common method that we're using for the component unit tests called `triggerGeneralObserver` check the [CodePen](https://github.com/PaulieScanlon/mdx-embed/blob/main/packages/mdx-embed/src/components/codepen/codepen.test.tsx) unit test for usage

**Review by PaulieScanlon — CHANGES_REQUESTED** [review]

File: packages/mdx-embed/src/components/twitch/twitch.test.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    observe: () => null,
+    unobserve: () => null,
+  });
+
+  beforeEach(() => {
```
Comment (PaulieScanlon) [inline packages/mdx-embed/src/components/twitch/twitch.test.tsx:14]: This can be removed

**Review by PaulieScanlon — CHANGES_REQUESTED** [review]

File: packages/mdx-embed/src/components/twitch/twitch.test.tsx (hunk 1 above)
Comment (ekafyi) [reply packages/mdx-embed/src/components/twitch/twitch.test.tsx:9]: Hi @PaulieScanlon,

Without lines 9 through 16 (`intersectionObserverMock` and the `beforeEach` function), only the first test passes; the remaining two tests fail with this error:

>  Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.

I added `intersectionObserverMock`, which seemed to fix that issue.

Would you prefer removing the second and third tests (so we only have one test, _'it renders the component when provided a twitchId'_); or having all three tests with `intersectionObserverMock`; or any other solution?

Comment (PaulieScanlon) [reply packages/mdx-embed/src/components/twitch/twitch.test.tsx:9]: @ekafyi thanks for this! Hmmm. that's interesting. Perhaps there's an issue with our mock observer. Leave it with me and i'll have a quick look and see if i can find the problem.

**Comment by PaulieScanlon** [comment]
@ekafyi You're absolutely right about needing to add the `beforeEach` method, thanks so much for catching this one, top job!

I'm not sure why single test instances pass but multiple test instances throw the exception probably must be a React thing 🤷‍♂️

In either case i've now exposed a new global `jest` method called `addIntersectionObserver` so you will have to pull the main branch again and rebase / merge the changes into your PR... apologies about that! 

Your test file should now look like this 👇

```javascript
import React from 'react';
import { render, screen, act } from '@testing-library/react';

import { Twitch } from './';

const PARENT_PROP = 'http://localhost:6006/';

describe('<Twitch />', () => {

  beforeEach(() => {
    (window as any).addIntersectionObserver();
  });

  test('it renders the component when provided a twitchId', () => {
    render(<Twitch parent={PARENT_PROP} twitchId="679474192" />);

    act(() => {
      (window as any).triggerGeneralObserver();
      return undefined;
    });

    expect(screen.getByTestId('twitch')).toBeDefined();
  });

  test('it renders the component when provided a channel', () => {
    render(<Twitch parent={PARENT_PROP} channel="jlengstorf" />);

    act(() => {
      (window as any).triggerGeneralObserver();
      return undefined;
    });

    expect(screen.getByTestId('twitch')).toBeDefined();
  });

  test('it renders the component when provided a collection', () => {
    render(<Twitch parent={PARENT_PROP} collection="eMybriFryRWpiQ" />);

    act(() => {
      (window as any).triggerGeneralObserver();
      return undefined;
    });

    expect(screen.getByTestId('twitch')).toBeDefined();
  });
});
```

**Review by PaulieScanlon — APPROVED** [review]

**Pull request merged**